### DISCUSSION TITLE
z_order over a Time Period

### DISCUSSION THREAD

**Discussion by tastyfrankfurt** [delta-io/delta-rs#2223] [Q&A]
Hey Team,

I want to enable z_order optimization with the delta-rs writer but i only want to do it on select time periods, as our dataset is so large its not good to do this over the entire dataset as this takes a lot longer to optimize. Please correct me if i have miss understood anything around z_order tasks.  Currently we use databricks to z_order similar to the example example below,
OPTIMIZE events
WHERE date >= current_timestamp() - INTERVAL 1 day
ZORDER BY (eventType)

Is some able to provide advice on how best to achieve this within the rust engine.

Many Thanks

**Comment by jijosg** [comment]
Upvote on this feature. Similar functionality is available in deltaspark
 `deltaTable.optimize().where("date='2021-11-18'").executeCompaction()`
